### PR TITLE
chore: unit test for LocalValue with some docs for the functions

### LINKED ISSUE #1: add LocalValue tests
all helper functions in LocalValue.ts should be tested

### PR DESCRIPTION
**Pull request by ranpafin**
Closes #1

### CODE REVIEW

File: src/LocalValue.ts [hunk 1]
```diff
@@ -150,6 +150,11 @@ const _reduceRight = <E, A, B>(
 // pipeables
 // -------------------------------------------------------------------------------------
 
+/**
+ * fold none/error/value state and return a value accordingly none in case of absent, error in case of invalid value otherwise
+ *
+ * @category constructors
```
Comment (fes300) [inline src/LocalValue.ts:156]: wouldn't folds be "destructors" based on your taxonomy?

File: src/LocalValue.ts [hunk 2]
```diff
@@ -165,6 +170,11 @@ export const fold = <E, A, B>(
   return onValue(fa.value)
 }
 
+/**
+ * fold none/error/value state and return a value accordingly. Error in case of invalid or absent value otherwise
+ *
+ * @category constructors
```
Comment (fes300) [inline src/LocalValue.ts:176]: wouldn't folds be "destructors" based on your taxonomy?

File: src/LocalValue.ts [hunk 3]
```diff
@@ -200,6 +220,11 @@ export const getOrElseW = <E, A, B>(defaultValue: Lazy<B>) => (
   return lv.value
 }
 
+/**
+ * Wrap a value into the type constructor.
+ *
+ * @category Applicative
```
Comment (fes300) [inline src/LocalValue.ts:226]: :love_letter:

File: src/LocalValue.ts [hunk 4]
```diff
@@ -208,12 +233,20 @@ export const chain = <E, A, B>(f: (a: A) => LocalValue<E, B>) => (
   return _chain(lv, f)
 }
 
+/**
+ * @category Functor
```
Comment (fes300) [inline src/LocalValue.ts:237]: :heart:

File: src/LocalValue.ts [hunk 5]
```diff
@@ -223,6 +256,11 @@ export const fromOption = <A>(o: Option<A>): LocalValue<never, A> =>
     ),
   )
 
+/**
+ * Transforms an Either to an LocalValue discarding the error.
```
Comment (fes300) [inline src/LocalValue.ts:260]: it is not actually discarding the value but wrapping it into an `invalid`

File: src/__tests__/localValue.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+    it("always return a valid value", () => {
+        const value = random.random().toString()
+
+        expect(of(value)).toEqual(valid(value))
+    })
+})
+
+describe("chain", () => {
+    it("extract the value if is valid, the localValue otherwise", () => {
+
```
Comment (fes300) [inline src/__tests__/localValue.spec.tsx:108]: this empty line shouldn't be here, did you run prettier?

**Review by fes300 — CHANGES_REQUESTED** [review]
wonderful :1st_place_medal: 
just a few minor changes

File: src/__tests__/localValue.spec.tsx (hunk 6 above)
Comment (ranpafin) [reply src/__tests__/localValue.spec.tsx]: actually no, i was under the impression the hook took care of that, running it manually fixed many things. I think we may need to review the hook as it gives (a false) greenlight on format.

File: src/LocalValue.ts (hunk 5 above)
Comment (ranpafin) [reply src/LocalValue.ts]: yes!

File: src/LocalValue.ts (hunk 2 above)
Comment (ranpafin) [reply src/LocalValue.ts]: 100% agree

File: src/LocalValue.ts (hunk 1 above)
Comment (ranpafin) [reply src/LocalValue.ts]: 100% agree

**Review by fes300 — APPROVED** [review]
💯

**Pull request merged**